### ISSUE TITLE
Keys are not consistently converted

### ISSUE THREAD

**Issue by jgeiger** [sds/mock_redis#170]
When using a key in Redis, 1 and "1" are equivalent (at least in Redis 3.x). While there are a few cases where this was fixed for some methods in #157 and #48, it should be a global fix for all key access.

I've recently run into this issue for lpush and lrange. I'll look into a PR, but this is more of a notice for anyone who may be seeing this issue in their code.

**Comment by meagar** [comment]
Similar problem for `hget`/`hset`, the results are very inconsistent:

r = MockRedis.new
r.hset(123, 'foo', 'bar') # => true
r.keys # => [] # expected [123] or ['123'], but not empty
r.hgetall(123) # => { 'foo' => 'bar' } # content is still queryable 
r.flushall # => "OK"
r.keys # => [] # as expected, the store should be empty
r.hgetall(123) # => { 'foo' => 'bar' } # numeric keys still map to values after `flushall`

**Comment by JeffLuckett** [comment]
There are also issues with using symbols as keys in Redis hashes.  See below:

```
{14:01}[2.5.1]~ ➭ irb
irb(main):001:0> require 'mock_redis'
=> true
irb(main):002:0> mr = MockRedis.new
=> #<MockRedis:0x00007f8540035b20 @options={:scheme=>"redis", :host=>"127.0.0.1", :port=>6379, :path=>nil, :timeout=>5.0, :password=>nil, :db=>0, :time_class=>Time}, @db=#<MockRedis::PipelinedWrapper:0x00007f853e9b7380 @db=#<MockRedis::TransactionWrapper:0x00007f853e9b7d80 @db=#<MockRedis::ExpireWrapper:0x00007f853e9b7e70 @db=#<MockRedis::MultiDbWrapper:0x00007f8540034c48 @db_index=0, @prototype_db=#<MockRedis::Database:0x00007f85400348b0 @base=#<MockRedis:0x00007f8540035b20 ...>, @data={}, @expire_times=[]>, @databases={0=>#<MockRedis::Database:0x00007f85400351e8 @base=#<MockRedis:0x00007f8540035b20 ...>, @data={}, @expire_times=[]>}>>, @transaction_futures=[], @multi_stack=[], @multi_block_given=false>, @pipelined_futures=[], @nesting_level=0>>
irb(main):003:0> mr.hset(:foo, :bar, :bas)
=> true
irb(main):004:0> mr.hgetall(:foo)
=> {"bar"=>"bas"}
irb(main):005:0> mr.data
=> {:foo=>{"bar"=>"bas"}}
irb(main):006:0> mr.del(:foo)
=> 0
irb(main):007:0> mr.data
=> {:foo=>{"bar"=>"bas"}}
```

Note the `del` operation didn't behave as expected, because `del` converts all keys to strings.  However, you can see in my `mr.data` call that the key is a `symbol`.

PR to follow.

**Comment by JeffLuckett** [comment]
This is addressed in this PR: https://github.com/sds/mock_redis/pull/173.

Until this PR is merged, and a new Gem version is cut, I'm working around this in my own project with the following monkey patch.

```ruby
# This library is only used in test.
if Rails.env.test?
  class MockRedis
    module HashMethods
      private

      def with_hash_at(key, &blk)
        version_check
        with_thing_at(key.to_s, :assert_hashy, proc { {} }, &blk)
      end

      def version_check
        return unless Gem.loaded_specs['mock_redis'].version > Gem::Version.new('0.21.0')
        Rails.logger.warn 'Check if PR (https://github.com/sds/mock_redis/pull/173) is merged, and part of current ' \
                          'version.  If so, remove this code.'
      end
    end
  end
end
```

**Comment by sds** [comment]
Fixed in #173.
